src/Reflection/ReflectionProperty.php

```diff
@@ -592,9 +671,123 @@ private function computeModifiers(PropertyNode $node): int
         $modifiers  = $node->isReadonly() ? ReflectionPropertyAdapter::IS_READONLY : 0;
         $modifiers += $node->isStatic() ? CoreReflectionProperty::IS_STATIC : 0;
         $modifiers += $node->isPrivate() ? CoreReflectionProperty::IS_PRIVATE : 0;
+        $modifiers += $node->isPrivateSet() ? ReflectionPropertyAdapter::IS_PRIVATE_SET_COMPATIBILITY : 0;
         $modifiers += $node->isProtected() ? CoreReflectionProperty::IS_PROTECTED : 0;
+        $modifiers += $node->isProtectedSet() ? ReflectionPropertyAdapter::IS_PROTECTED_SET_COMPATIBILITY : 0;
         $modifiers += $node->isPublic() ? CoreReflectionProperty::IS_PUBLIC : 0;
+        $modifiers += ($node->flags & ReflectionPropertyAdapter::IS_FINAL_COMPATIBILITY) === ReflectionPropertyAdapter::IS_FINAL_COMPATIBILITY ? ReflectionPropertyAdapter::IS_FINAL_COMPATIBILITY : 0;
+        $modifiers += ($node->flags & Modifiers::ABSTRACT) === Modifiers::ABSTRACT ? ReflectionPropertyAdapter::IS_ABSTRACT_COMPATIBILITY : 0;
 
+        /** @phpstan-ignore return.type */
         return $modifiers;
     }
+
+    private function computeImmediateVirtual(PropertyNode $node): bool
+    {
+        if ($node->hooks === []) {
+            return false;
+        }
+
+        $setHook = null;
+
+        foreach ($node->hooks as $hook) {
+            if ($hook->name->name === 'set') {
+                $setHook = $hook;
+                break;
+            }
+        }
+
+        if ($setHook === null) {
+            return true;
+        }
+
+        $setHookBody = $setHook->getStmts();
+        assert($setHookBody !== null);
+
+        // Short syntax
+        if (count($setHookBody) === 1 && $setHookBody[0] instanceof Node\Stmt\Return_) {
+            return false;
+        }
+
+        $visitor   = new FindingVisitor(static fn (Node $node): bool => $node instanceof Node\Expr\Assign);
+        $traverser = new NodeTraverser($visitor);
+        $traverser->traverse($setHookBody);
+
+        foreach ($visitor->getFoundNodes() as $assigNode) {
+            assert($assigNode instanceof Node\Expr\Assign);
+            $variableToAssign = $assigNode->var;
+
+            if (
+                $variableToAssign instanceof Node\Expr\PropertyFetch
+                && $variableToAssign->var instanceof Node\Expr\Variable
+                && $variableToAssign->var->name === 'this'
+                && $variableToAssign->name instanceof Node\Identifier
+                && $variableToAssign->name->name === $this->name
+            ) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /** @return array{get?: ReflectionMethod, set?: ReflectionMethod} */
+    private function createImmediateHooks(PropertyNode $node): array
+    {
+        $hooks = [];
+
+        foreach ($node->hooks as $hook) {
+            $hookName = $hook->name->name;
+            assert($hookName === 'get' || $hookName === 'set');
```

Comment:
This assert will work for now, but may fail if hooks expand in scope (`isset`, for example)

Reply:
Alrighty, will glance over it, for now :)